client/client-multi/client-cpp/impl/KaaClient.cpp

```diff
@@ -270,44 +270,191 @@ void KaaClient::setDefaultConfiguration()
 #endif
 }
 
-IProfileManager& KaaClient::getProfileManager()
-{
-    return *profileManager_;
+
+void KaaClient::setProfileContainer(ProfileContainerPtr container) {
+    profileManager_->setProfileContainer(container);
 }
 
-IConfigurationPersistenceManager& KaaClient::getConfigurationPersistenceManager()
-{
+ISerializedProfileContainerPtr KaaClient::getSerializedProfileContainer() {
+    return profileManager_->getSerializedProfileContainer();
+}
+
+
+void KaaClient::setConfigurationStorage(IConfigurationStoragePtr storage) {
 #ifdef KAA_USE_CONFIGURATION
-    return *configurationPersistenceManager_;
+    configurationPersistenceManager_->setConfigurationStorage(storage);
 #else
-    throw KaaException("Failed to retrieve ConfigurationPersistenceManager. Configuration subunit is disabled");
+    throw KaaException("Failed to set configuration storage. Configuration subunit is disabled");
```

Comment:
Please, use 'subsystem' instead of 'subunit' to be consistent with the public documentation.